cogs/fun.py

```diff
@@ -1,16 +1,272 @@
 import random
 import textwrap
-
+import urllib
+import typing as t
+import aiohttp
+import os
 import discord
-from discord import Color, Embed
-from discord.ext.commands import BadArgument, Bot, BucketType, Cog, Context, command, cooldown
+from .utils import constants
+from random import choice, randint
+from discord import Color, Embed, Message
+from discord.ext.commands import BadArgument, Bot, BucketType, Cog, Context, command, cooldown, is_nsfw, errors
+
+from cogs.utils.errors import ServiceError
+
+from .endpoints.endpoints import nekos
+import nekos as n
+
+
+file = open("assets" + os.path.sep + "excuses.txt", "r", encoding="utf-8")
+excuses = file.readlines()
+file.close()
 
 
 class Fun(Cog):
     """This is a cog for simple fun commands."""
 
     def __init__(self, bot: Bot) -> None:
         self.bot = bot
+        self.user_agent = {"User-Agent": "HotWired"}
+        self.dadjoke = {
+            "User-Agent": "HotWired",
+            "Accept": "text/plain",
+        }
+
+    @command()
+    async def catfact(self, ctx: Context) -> None:
+        """Sends a random cat fact"""
+        async with aiohttp.ClientSession() as session:
+            async with session.get("https://cat-fact.herokuapp.com/facts") as response:
+                self.all_facts = await response.json()
+        fact = choice(self.all_facts["all"])
+        await ctx.send(embed=Embed(title="Did you Know?", description=fact["text"], color=0x690E8))
+
+    @command()
+    async def textcat(self, ctx: Context) -> None:
+        """Sends a random textcat"""
+        try:
+            embed = Embed(color=0x690E8)
+            embed.set_image(url=n.textcat())
+            await ctx.send(embed=embed)
+        except Exception:
```

Comment:
You shouldn't except for bare `Exception`, please specify the exact error which occurs here.
Same goes for lines 61 and 71.

Reply:
done